src/Platforms.jl

```diff
@@ -14,3 +16,201 @@ Pkg.BinaryPlatforms.triplet(::AnyPlatform) = "any"
 # Fallback on x86_64-linux-musl, but this shouldn't really matter in practice.
 Pkg.BinaryPlatforms.arch(::AnyPlatform) = arch(Linux(:x86_64, libc=:musl))
 Base.show(io::IO, ::AnyPlatform) = print(io, "AnyPlatform()")
+
+## The extended platforms, to represent platforms beyond the standard ones in
+## Pkg.BinaryPlatforms.
+
+struct ExtendedPlatform{P<:Platform} <: Platform
+    p::P
+    ext::Dict{String,String}
+end
+ExtendedPlatform(p::Platform; kwargs...) =
+    ExtendedPlatform(p, Dict(string(k) => string(v) for (k, v) in pairs(kwargs)))
+
+Pkg.BinaryPlatforms.platform_name(::ExtendedPlatform) = "ExtendedPlatform"
+Pkg.BinaryPlatforms.arch(ep::ExtendedPlatform) = arch(ep.p)
+Pkg.BinaryPlatforms.libc(ep::ExtendedPlatform) = libc(ep.p)
+Pkg.BinaryPlatforms.call_abi(ep::ExtendedPlatform) = call_abi(ep.p)
+Pkg.BinaryPlatforms.compiler_abi(ep::ExtendedPlatform) = compiler_abi(ep.p)
+Pkg.BinaryPlatforms.triplet(ep::ExtendedPlatform) =
+    triplet(ep.p) * join(["-$(k)+$(ep.ext[k])" for k in sort(collect(keys(ep.ext)))])
+
+# This function also strips out the extra features of the type
+abi_agnostic(ep::ExtendedPlatform) = abi_agnostic(ep.p)
+
+function Base.show(io::IO, ep::ExtendedPlatform)
+    print(io, BinaryPlatforms.platform_name(ep), "(", ep.p)
+    length(ep.ext) > 0 && print(io, ";")
+    join(io, [" $(k)=$(repr(v))" for (k, v) in ep.ext], ",")
+    print(io, ")")
+end
+
+# Vendor Pkg.BinaryPlatforms.platform_key_abi, with a different name.  The
+# difference with the original version is that if the triplet can't be parse we
+# return `nothing`, wthout issuing a warning
+function tryparse_standard_platform(machine::AbstractString)
```

Comment:
It's up to you whether the performance hit of disabling logging is worth it, but I am always loathe to duplicate code, since I've already committed that sin many times in my life.  ;)

```
 using Logging, Pkg.BinaryPlatforms
function tryparse_standard_platform(machine::AbstractString)
    p = with_logger(NullLogger()) do
        BinaryPlatforms.platform_key_abi("dsgaeg")
    end
    if isa(p, UnknownPlatform)
        return nothing
    end
    return p
end
```